src/book/mod.rs

```diff
@@ -133,69 +131,44 @@ impl MDBook {
 
         debug!("[fn]: init");
 
-        if !self.config.get_root().exists() {
-            fs::create_dir_all(&self.config.get_root()).unwrap();
-            info!("{:?} created", &self.config.get_root());
-        }
-
         {
-
-            if !self.get_destination().exists() {
-                debug!("[*]: {:?} does not exist, trying to create directory", self.get_destination());
-                fs::create_dir_all(self.get_destination())?;
-            }
-
-
-            if !self.config.get_source().exists() {
-                debug!("[*]: {:?} does not exist, trying to create directory", self.config.get_source());
-                fs::create_dir_all(self.config.get_source())?;
+            let root = self.config.get_root();
+            let dest = self.get_destination();
+            let src = self.config.get_source();
+
+            let necessary_folders = &[root, dest, src];
+            
+            for folder in necessary_folders {
+                if !folder.exists() {
+                    fs::create_dir_all(folder)?;
+                    debug!("{} created", folder.display());
+                }
             }
 
-            let summary = self.config.get_source().join("SUMMARY.md");
+            let summary = src.join("SUMMARY.md");
 
             if !summary.exists() {
+                debug!("[*]: Creating SUMMARY.md");
 
-                // Summary does not exist, create it
-                debug!("[*]: {:?} does not exist, trying to create SUMMARY.md", &summary);
                 let mut f = File::create(&summary)?;
 
-                debug!("[*]: Writing to SUMMARY.md");
-
                 writeln!(f, "# Summary")?;
-                writeln!(f, "")?;
+                writeln!(f)?;
                 writeln!(f, "- [Chapter 1](./chapter_1.md)")?;
             }
-        }
 
-        // parse SUMMARY.md, and create the missing item related file
-        self.parse_summary()?;
-
-        debug!("[*]: constructing paths for missing files");
```

Comment:
That's interesting. I've actually found that feature to be annoying in the past because if you accidentally make a typo in your `SUMMARY.md` or run `mdbook build` in the wrong directory it'll silently succeed and generate stubs for the files. When really it should let you know your `SUMMARY.md` is incorrect so you can manually fix it.

If you're running `mdbook init`, doesn't that imply that you don't already have a `SUMMARY.md` file though? So it doesn't really make sense to look for one in that case, when `init` should just be generating a stub book to get you going... That was my logic anyway. But it's definitely open to discussion.

Reply:
What about adding a switch to the command line program so you can do `mdbook build --create-stubs` to tell `mdbook` to parse the `SUMMARY.md` then create dummy files for any chapters which don't exist? 

I just feel creating the stubs without the user explicitly asking for it is a bit too implicit/magic, which is why I left out the feature when merging the `Book` stuff in.

Reply:
>if you accidentally make a typo in your SUMMARY.md or run mdbook build in the wrong directory it'll silently succeed and generate stubs for the files.

Maybe we should remove this behaviour from the `build` command but leave it in the `init`.

>If you're running mdbook init, doesn't that imply that you don't already have a SUMMARY.md file though? 

Not necessarily, I think of `init` as a way to generate the missing files. In that perspective it makes sense. But as you pointed out, it may not align with everyones intuition.

I am ok with having this as an opt-in with an extra argument. I'm not sure about the name of the argument though `--create-chapters`, `--gen-chapters`, ...

@budziq do you have an opinion about this?

Reply:
> I am ok with having this as an opt-in with an extra argument. I'm not sure about the name of the argument though --create-chapters, --gen-chapters, ...
budziq do you have an opinion about this?

@azerupi sorry for lagging with the review (I'll try to make some time on Monday).
In regard to automatic generation of chapters. In terms of an API, I would suggest additional argument for a builder type. And in terms of actual mdbook binary I would go with additional option as you suggested.

Reply:
>sorry for lagging with the review

No worries, do not feel forced to review anything if you don't have the time :)

>And in terms of actual mdbook binary I would go with additional option as you suggested.

Ok, good! I think everyone is in agreement here, so I propose that we make this opt-in behind a CLI flag both available for `build` and `init`

Reply:
Sounds good. I have a feeling the `MDBook` helper struct might need a bit of restructuring at some point (perhaps converting it into a builder?) so auto-generating missing bits would just be a case of setting a flag or calling a method.

For now should I just add a boolean flag to the `MDBook` and a method for setting that flag, then work it into the `build()` and `init()` methods?

Reply:
Yes, sounds good :)